packages/wallet/src/cip30.ts

```diff
@@ -49,9 +50,23 @@ export type SubmitTxCallbackParams = {
   data: Cardano.Tx;
 };
 
-export type CallbackConfirmation = (
-  args: SignDataCallbackParams | SignTxCallbackParams | SubmitTxCallbackParams
-) => Promise<boolean>;
+// Optional callback
+export type GetCollateralCallbackParams = {
+  type: Cip30ConfirmationCallbackType.GetCollateral;
+  data: {
+    amount: Cardano.Lovelace;
+    utxos: Cardano.Utxo[];
+  };
+};
+
+type GetCollateralCallback = (args: GetCollateralCallbackParams) => Promise<Cardano.Utxo[]>;
+
+export type CallbackConfirmation = {
```

Comment:
Just realized this makes it a breaking change, please reword the commit header as `feat(wallet)!: add optional callback for getCollateral` and mention the refactor in commit body

Reply:
Added this change, had to format the body across 2 lines, please let me know if I should reword to condense to a single line.